apps/prairielearn/assets/scripts/instructorFileEditorClient.ts

```diff
@@ -57,6 +88,18 @@ class InstructorFileEditor {
     });
 
     this.editor.getSession().on('change', () => this.onChange());
+
+    if (aceMode === 'ace/mode/json') {
+      this.editor.getSession().setTabSize(2);
+      document
+        .querySelector<HTMLButtonElement>('.js-reformat-file')
+        ?.addEventListener('click', () =>
+          this.reformatJSONFile().catch((err) => {
+            // The document probably has invalid JSON syntax. Just do nothing.
+            console.error(err);
```

Comment:
Nit: when there is invalid JSON syntax this works as expected, but there is no user feedback, which may be confusing. Do you want to provide a user-facing error (e.g., a popover in the "Reformat" button) so the user knows that this failed? Won't block on this, though.

Reply:
We could use a [Bootstrap toast](https://getbootstrap.com/docs/4.6/components/toasts/) for this. How does that sound to you?

Reply:
That would be a new use, but it's neat, and I can see how it can be useful in other scenarios. I was thinking more of a popover that fades out after a bit, similar to the clipboard "Copied" used in:
https://github.com/PrairieLearn/PrairieLearn/blob/9a739c490b98cce212baaef084af4aa3504a92e9/apps/prairielearn/assets/scripts/lib/clipboardPopover.ts#L7-L10

I trust your judgement...

Reply:
FWIW I've always thought that was an abuse of popovers 🙃 I'll give toasts a shot.

Reply:
Done in d62dbc3:

<img width="1469" alt="Screenshot 2024-07-22 at 11 35 00" src="https://github.com/user-attachments/assets/6d7a21c0-4e80-46ce-aeed-335ba52077e1">